src/NuGet.Core/NuGet.Configuration/Settings/Items/CredentialsItem.cs

```diff
@@ -190,6 +190,11 @@ internal override SettingBase Clone()
                 newSetting.SetOrigin(Origin);
             }
 
+            foreach(var attr in Attributes)
```

Comment:
Check `SettingsElement.Attributes`.  It is incorrectly defined as:

    internal IReadOnlyDictionary<string, string> Attributes => new ReadOnlyDictionary<string, string>(MutableAttributes);

It should be:

    internal IReadOnlyDictionary<string, string> Attributes { get; } = new ReadOnlyDictionary<string, string>(MutableAttributes);

Reply:
Good catch! This seems unrelated to the work on this PR. Should I do this change in this PR or create a new one just for this?

Reply:
Either way.